x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -55,7 +55,9 @@ type Aggregator struct {
 	tooManyGroupsLogger *logp.Logger
 
 	mu               sync.RWMutex
-	active, inactive *metrics
+	active, inactive map[time.Duration]*metrics
```

Comment:
Given that we iterate over the intervals in order, would it be worth making these `[]*metrics`?

Reply:
I suppose we could, would you prefer that?

Reply:
I don't have a very strong opinion on this - slight preference because it'll be slightly faster. Not sure if it's worth it. If it makes things harder to read then don't worry about it.

Reply:
I looked into making the change but in my opinion it made the code slightly less readable and cryptic. I'm happy to revisit if you feel it would be better to move it to a slice.

Reply:
Totally fine, thanks for looking into it.